docs/feature-store/sources-targets.md

```diff
@@ -0,0 +1,67 @@
+| [mlrun.datastore.CSVTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.CSVTarget)        | Writes events to a CSV file (offline target).                                                                       | Y      | Y     | Y      |
+| [mlrun.datastore.KafkaTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.KafkaTarget)    | Writes all incoming events into a Kafka stream (online target).                                                    | Y | N | N |
+| [mlrun.datastore.NoSqlTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.NoSqlTarget)    | The default online target. Persists the data in V3IO table to its associated storage by key (online target).       | Y      | Y     | Y      |
+| mlrun.datastore.RedisNoSqlTarget                                                                              | Persists the data in Redis table to its associated storage by key (online target).                                 | Y      | Y     | N      |
+| mlrun.datastore.SqlTarget                                                                                     | The default offline target. Persists the data in SQL table to its associated storage by key (offline target).      | Y      | N     | Y      |
+| [mlrun.datastore.ParquetTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.ParquetTarget)| The Parquet target storage driver, used to materialize feature set/vector data into parquet files (online target). | Y      | Y     | Y      |
+| [mlrun.datastore.StreamTarget](https://storey.readthedocs.io/en/latest/api.html#storey.targets.StreamTarget)  | Writes all incoming events into a V3IO stream (offline target).                                                    | Y      | N     | N      |
+
+## ParquetTarget
+
+### Partitioning
+
+When writing data to a {py:meth}`~mlrun.datastore.ParquetTarget`, you can use partitioning. Partitioning organizes data 
+in Parquet files by dividing large data sets into smaller and more manageable pieces. The data is divided
+into separate files according to specific criteria, for example: date, time, or specific values in a column.
+Partitioning, when configured correctly, improves read performance by reducing the amount of data that needs to be 
+processed for any given function, for example, when reading back a limited time range with `get_offline_features()`.
+
+```{note} 
+When partitioning with a pandas engine, the `ParquetTarget(path="my_file.parquet")` writes to a single Parquet file. 
+`ingest()` calls `df.to_parquet()` once, with all the data. A single `df.to_parquet()` call cannot write to more than 1024 partitions.
+```
+storey processes the data row by row (as a streaming engine, it doesn't get all the data up front, so it needs to process row by row). 
+These rows are batched together according to the partitions defined, and they are 
+written to each partition separately. (Therefore, storey does not have the 1024 limitation.)
```

Comment:
When using the pandas engine for ingestion, pandas incur a maximum limit of 1024 partitions on each ingestion.
If the data being ingested spans over more than 1024 partitions, the ingestion will fail.
Decreasing the number of partitions can be achieved by filtering by time or increasing the `time_partitioning_granularity`

Reply:
@assaf758 what do you mean by time filtering? Isn't that <time_partitioning_granularity>

Reply:
time-filter is a feature of sources - for example ParquetSource has start_filter/end_filter.
time_partitioning_granularity is the granularity (the time range that each partition holds)